src/Cryptol/TypeCheck/TCon.hs

```diff
@@ -1,174 +1,86 @@
-
-
--- | This list should contain all user-visible built-in types.
-primTyList :: [ PrimTy ]
-primTyList =
-  [ -- Value type constructors -------------------------------------------------
-
-    tPrefix "inf"             TC TCInf
-    "A numeric type representing infinity."
-
-  , tPrefix "Bit"             TC TCBit
-    "The type of boolean values."
-
-  , tPrefix "Integer"         TC TCInteger
-    "The type of unbounded integers."
-
-  , tPrefix "Z"               TC TCIntMod
-    "'Z n' is the type of integers, modulo 'n'."
-
-
-    -- Predicate constructors --------------------------------------------------
-
-  , tInfix "=="                PC PEqual      (n 20)
-    "Assert that two numeric types are equal."
-
-  , tInfix "!="                PC PNeq        (n 20)
-    "Assert that two numeric types are different."
-
-  , tInfix ">="                PC PGeq        (n 30)
-    "Assert that the first numeric type is larger than, or equal to the second."
-
-  , tPrefix "fin"              PC PFin
-    "Assert that a numeric type is a proper natural number (not 'inf')."
-
-  , tPrefix "Zero"             PC PZero
-    "Value types that have a notion of 'zero'."
-
-  , tPrefix "Logic"            PC PLogic
-    "Value types that support logical operations."
-
-  , tPrefix "Arith"            PC PArith
-    "Value types that support arithmetic."
-
-  , tPrefix "Cmp"              PC PCmp
-    "Value types that support unsigned comparisons."
-
-  , tPrefix "SignedCmp"        PC PSignedCmp
-    "Value types that support signed comparisons."
-
-  , tPrefix "Literal"         PC PLiteral
-    "'Literal n a' asserts that type 'a' contains the number 'n'."
-
-
-
-    -- Type functions ------------------------------------------------
-
-  , tInfix  "+"                TF TCAdd       (l 80)
-    "Add numeric types."
-
-  , tInfix  "-"                TF TCSub       (l 80)
-    "Subtract numeric types."
-
-  , tInfix  "*"                TF TCMul       (l 90)
-    "Multiply numeric types."
-
-  , tInfix  "/"                TF TCDiv       (l 90)
-    "Divide numeric types, rounding down."
-
-  , tInfix  "%"                TF TCMod       (l 90)
-    "Remainder of numeric type division."
-
-  , tInfix  "^^"               TF TCExp       (r 95)
-    "Exponentiate numeric types."
-
-  , tPrefix "width"            TF TCWidth
-    "The number of bits required to represent the value of a numeric type."
-
-  , tPrefix "min"              TF TCMin
-    "The smaller of two numeric types."
-
-  , tPrefix "max"              TF TCMax
-    "The larger of two numeric types."
-
-  , tInfix  "/^"               TF TCCeilDiv    (l 90)
-    "Divide numeric types, rounding up."
-
-  , tInfix  "%^"               TF TCCeilMod    (l 90)
-    "How much we need to add to make a proper multiple of the second argument."
-
-  , tPrefix "lengthFromThenTo" TF TCLenFromThenTo
-    "The length of an enumeration."
-  ]
+-- | This is used for pretty prinitng.
```

Comment:
What is difficult about getting fixity information from the Prelude?

Reply:
It's quite annoying.  It's two things, either of which could probably be fixed somehow:  a) the pretty printer does not have access to additional information, such as an environment of fixities, b) alternatively the fixities could be stored directly on the constructors, but that makes it difficult to make new constructors, as now you have to know their fixities (e.g., look at `tMul` in `Cryptol.TypeCheck.Type`).

Another option is the following "hack":  we already use TH to bake in the Prelude in Cryptol, but currently it is just stored as a string.  We could, however, parse it further, and generate the fixity table.

Reply:
I'll leave this as is for the moment, as I think fixing it requires some changes that are otherwise not very related to this pull request.    I made ticket #634 to help us remember to do it.